distribution/src/bin/elasticsearch

```diff
@@ -20,6 +20,27 @@ if [ -z "$ES_TMPDIR" ]; then
   ES_TMPDIR=`"$JAVA" -cp "$ES_CLASSPATH" org.elasticsearch.tools.launchers.TempDirectory`
 fi
 
+# get keystore password before setting java options to avoid
+# conflicting GC configurations for the keystore tools
+unset KEYSTORE_PASSWORD
+KEYSTORE_PASSWORD=
+if ! echo $* | grep -E -q '(^-h |-h$| -h |--help$|--help |^-V |-V$| -V |--version$|--version )' \
+    && "`dirname "$0"`"/elasticsearch-keystore has-passwd --silent
+then
+  if [ -e "$ES_KEYSTORE_PASSPHRASE_FILE" ] ; then
```

Comment:
I don't believe that will work for the Systemd case, but I'll take another look at the systemd docs and report back.

Reply:
OK, this is doable, but in a hacky way. We have to do something like this in our service defintion:

```
Environment=EKS_PASS=/etc/elasticsearch/eks

...

ExecStart=/bin/sh -c 'if [ -n "$EKS_PASS" ] ; then exec /usr/share/elasticsearch/bin/elasticsearch -p ${PID
_DIR}/elasticsearch.pid --quiet <"$EKS_PASS" ; else exec /usr/share/elasticsearch/bin/elasticsearch -p ${PI
D_DIR} --quiet ; fi'
```

It might make sense to replace this with a little systemd-entrypoint wrapper script.

As a side note: The Vagrant box I'm using for Centos 7 testing has Systemd version 219. Version 243 has some additional features for setting StandardInput but I decided not to pursue them: https://twitter.com/adamcatino/status/1179775966744236033

Reply:
OK, I think I've got this one. It's all in one commit so it'll be easy to roll back if I've over-engineered it.